src/containers/add-attachments/AddAttachments.tsx

```diff
@@ -1,30 +1,41 @@
 import { ChangeEvent, FC, useCallback, useState } from 'react'
 import { useTranslation } from 'react-i18next'
+import SearchIcon from '@mui/icons-material/Search'
 import Box from '@mui/material/Box'
 import Typography from '@mui/material/Typography'
-import SearchIcon from '@mui/icons-material/Search'
 
-import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
 import AppButton from '~/components/app-button/AppButton'
 import EnhancedTable from '~/components/enhanced-table/EnhancedTable'
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
 import { useModalContext } from '~/context/modal-context'
-import useBreakpoints from '~/hooks/use-breakpoints'
 import useSort from '~/hooks/table/use-sort'
 import useAxios from '~/hooks/use-axios'
+import useBreakpoints from '~/hooks/use-breakpoints'
 import { attachmentService } from '~/services/attachment-service'
 
-import { ajustColumns } from '~/utils/helper-functions'
+import { defaultResponses } from '~/constants'
 import {
-  initialSort,
   columns,
+  initialSort,
   removeColumnRules
 } from '~/containers/add-attachments/AddAttachments.constants'
-import { defaultResponses } from '~/constants'
 import { styles } from '~/containers/add-attachments/AddAttachments.styles'
 import { Attachment, ButtonVariantEnum } from '~/types'
+import { ajustColumns } from '~/utils/helper-functions'
 
-const AddAttachments: FC = () => {
+interface AddAttachmentsProps {
+  attachments: Attachment[]
+  onAddAttachments: (attachments: Attachment[]) => void
+}
+
+const AddAttachments: FC<AddAttachmentsProps> = ({
+  attachments,
+  onAddAttachments
+}) => {
   const [inputValue, setInputValue] = useState<string>('')
+  const [selectedAttachments, setSelectedAttachments] = useState<Attachment[]>(
+    attachments.length ? attachments : []
```

Comment:
do we need this check, maybe attachments will be an empty array as a default response?

Reply:
yeap, it would looks good so i have added default value